src/App.tsx

```diff
@@ -1,24 +1,155 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
+import { FilterTodos } from './types/FilterTodos';
+import { getTodos, createTodo, deleteTodo } from './api/todos';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+const USER_ID = 6550;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(Error.NONE);
+  const [filterBy, setFilterBy] = useState(FilterTodos.ALL);
+  const [activeInput, setActiveInput] = useState(true);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [idUpdating, setIdUpdating] = useState([0]);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      try {
+        setErrorMessage(Error.NONE);
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setErrorMessage(Error.LOAD);
+      }
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const countTodos = todos.length;
+
+  const activeTodos = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const visibleTodos: Todo[] = useMemo(() => {
+    switch (filterBy) {
+      case FilterTodos.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case FilterTodos.COMPLTED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, filterBy]);
+
+  const addTodo = (title: string) => {
+    if (!title.trim()) {
+      setErrorMessage(Error.TITLE);
+
+      return;
+    }
+
+    const createNewTodo = async () => {
+      try {
+        setErrorMessage(Error.NONE);
+
+        const newTodo = {
+          userId: USER_ID,
+          title,
+          completed: false,
+        };
+
+        setActiveInput(false);
+        setTempTodo({ ...newTodo, id: 0 });
+        const createdTodo = await createTodo(USER_ID, newTodo);
+
+        setTodos(prevTodos => [...prevTodos, createdTodo]);
+      } catch (error) {
+        setErrorMessage(Error.ADD);
+      } finally {
+        setTempTodo(null);
+        setActiveInput(true);
+      }
+    };
+
+    createNewTodo();
```

Comment:
there is no need to create separate function inside the `addTodo`
```suggestion
    try {
      setErrorMessage(Error.NONE);

      const newTodo = {
        userId: USER_ID,
        title,
        completed: false,
      };

      setActiveInput(false);
      setTempTodo({ ...newTodo, id: 0 });
      const createdTodo = await createTodo(USER_ID, newTodo);

      setTodos(prevTodos => [...prevTodos, createdTodo]);
    } catch (error) {
      setErrorMessage(Error.ADD);
    } finally {
      setTempTodo(null);
      setActiveInput(true);
    }
```